build.go

```diff
@@ -46,40 +58,58 @@ func PersistSegmentBase(sb *SegmentBase, path string) error {
 		_ = os.Remove(path)
 	}
 
-	br := bufio.NewWriter(f)
-
-	_, err = br.Write(sb.mem)
+	_, err = persistSegmentBaseToWriter(sb, f)
 	if err != nil {
 		cleanup()
 		return err
 	}
 
-	err = persistFooter(sb.numDocs, sb.storedIndexOffset, sb.fieldsIndexOffset, sb.docValueOffset,
-		sb.chunkMode, sb.memCRC, br)
+	err = f.Sync()
 	if err != nil {
 		cleanup()
 		return err
 	}
 
-	err = br.Flush()
+	err = f.Close()
 	if err != nil {
 		cleanup()
 		return err
 	}
 
-	err = f.Sync()
+	return err
+}
+
+type bufWriter struct {
```

Comment:
I thought maybe we could use the existing CountHashWriter, but I guess the cost of computing unnecessary crc32 is too high....